com.unity.render-pipelines.high-definition/Runtime/Water/HDRenderPipeline.WaterSystem.cs

```diff
@@ -0,0 +1,786 @@
+using UnityEngine.Experimental.Rendering;
+using UnityEngine.Experimental.Rendering.RenderGraphModule;
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    // Enum that defines the sets of resolution at which the water simulation can be evaluated
+    public enum WaterSimulationResolution
+    {
+        Low64 = 64,
+        Medium128 = 128,
+        High256 = 256,
+        Ultra512 = 512
+    }
+
+    internal class WaterSimulationResources
+    {
+        // Overall time that has passed since Unity has been initialized
+        private float m_Time = 0;
+
+        // Current simulation time (used to compute the dispersion of the Phillips spectrum)
+        public float simulationTime = 0;
+
+        // Delta time of the current frame
+        public float deltaTime = 0;
+
+        // Texture that holds the Phillips spectrum
+        public RTHandle phillipsSpectrumBuffer = null;
+
+        // Texture that holds the displacement buffers
+        public RTHandle displacementBuffer = null;
+
+        // Texture that holds the additional data buffers (normal + foam)
+        public RTHandle additionalDataBuffer = null;
+
+        // Resolution at which the water system is ran
+        public int simulationResolution = 0;
+
+        // The number bands that we will be running the simulation at
+        public int numBands = 0;
+
+        // The wind speed, orientation and weight used to evaluate the Phillips spectrum
+        public float windSpeed = 0;
+        public float windOrientation = 0;
+        public float windAffectCurrent = 0;
+
+        // Value that defines the patch sizes of the bands (up to 4)
+        public Vector4 patchSizes = Vector4.zero;
+
+        // Value that defines the wind speed that is applied to each patch (up to 4)
+        public Vector4 patchWindSpeed = Vector4.zero;
+
+        // Function that allocates the resources and keep track of the resolution and number of bands
+        public void AllocateSmmulationResources(int simulationRes, int nbBands)
+        {
+            // Keep track of the values that constraint the texture allocation.
+            simulationResolution = simulationRes;
+            numBands = nbBands;
+
+            // Allocate the buffers
+            phillipsSpectrumBuffer = RTHandles.Alloc(simulationResolution, simulationResolution, numBands, dimension: TextureDimension.Tex2DArray, colorFormat: GraphicsFormat.R16G16_SFloat, enableRandomWrite: true, wrapMode: TextureWrapMode.Repeat);
```

Comment:
sanity check, is this always allocated even if water is disabled?

Reply:
Only allocated if the system is on